cmd/delete-all.go

```diff
@@ -33,23 +35,75 @@ var DeleteAll = &cobra.Command{
 			return fmt.Errorf("resource %s doesn't support DELETE", args[0])
 		}
 
-		min := resource.DeleteEntityInfo.MinResources
+		allParentEntityIds, err := getParentIds(context.Background(), resource)
 
-		delName := resource.SingularName
-
-		ids, err := getPage(args[0])
 		if err != nil {
-			return fmt.Errorf("problem getting page of ids for resource %s", args[0])
+			return fmt.Errorf("could not retrieve parent ids for for resource %s, error: %w", resource.PluralName, err)
+		}
+
+		if len(allParentEntityIds) == 1 {
+			log.Infof("Resource %s is a top level resource need to scan only one path to delete all resources", resource.PluralName)
+		} else {
+			log.Infof("Resource %s is not a top level resource, need to scan %d paths to delete all resources", resource.PluralName, len(allParentEntityIds))
 		}
 
-		for len(ids) > min {
-			delPage(delName, ids)
-			ids, err = getPage(args[0])
+		for _, parentEntityIds := range allParentEntityIds {
+			for ; ; {
```

Comment:
Can check whether previous version of ids matches the current set, if we've made no progress abort.